File: vue_acimage_web/mock/topic.js
import Mock from 'mockjs'
import { Code } from '@/utils/result.js'

Mock.mock('/api/community/topics/recentHot', 'get', {
	code: Code.OK,
	'data|4': [{
			id: '@id()',
			'title|5': '@cname()',
			starCount: 888,
			pageView: 88888,
			createTime: '@date()',
			user: {
				username: '@cname()',
				photoUrl: ''
			},
			coverImageUrl: 'test/test1.jpeg',
			categoryId:1,
		},
		{
			id: '@id()',
			'title|15': '@cname()',
			starCount: 888,
			pageView: 88888,
			createTime: '2022-2-22 22:22:22',
			user: {
				username: '@cname()',
				photoUrl: ''
			},
			coverImageUrl: 'topicImage/2022/12/02/1573240094587424768',
			categoryId:1,
		},
	]
})

Mock.mock('/api/community/topics/recommend', 'get', {
	code: Code.OK,
	'data|2': [{
			id: '@integer()',
			'title|5': '@cname()',
			commentCount:123,
			starCount: 888,
			pageView: 88888,
			createTime: '@date()',
			user: {
				username: '@cname()',
				photoUrl: ''
			},
			coverImageUrl: 'test/test1.jpeg',
		},
		{
			id: '@integer()',
			'title|15': '@cname()',
			commentCount:123,
			starCount: 888,
			pageView: 88888,
			createTime: '2022-2-22 22:22:22',
			user: {
				username: '刚刚刚刚刚刚刚刚',
				photoUrl: ''
			},
			coverImageUrl: 'topicImage/2022/12/02/1573240094587424768',
			categoryId:1,
		},
	]
})

Mock.mock('/api/community/topics/pageRecentTopics/1', 'get', {
	code: Code.OK,
	data: {
		'dataList|5': [{
			id: '@integer()',
			userId: 999,
			'title|10': "@cname()",
			'content': `不知道@cparagraph()`,
			activityTime: '2022-2-22 2:22:22',
			starCount: 666,
			pageView: 777,
			commentCount: 888,
			coverImageUrl: 'static/image/user-rank-header.jpg',
			categoryId:1,
			images: [{
				id: 0
			}],
			user: {
				photoUrl: '',
				'username|2': "@cname()"
			}
		}, ],
		totalCount: 10,
	}

})

File: acimage_common/src/main/java/com/acimage/common/model/domain/community/Topic.java
package com.acimage.common.model.domain.community;


import com.acimage.common.model.domain.user.User;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class Topic {
    public static final int CONTENT_MIN = 4;
    public static final int CONTENT_MAX = 500;
    public static final String CONTENT_VALIDATION_MSG= "内容长度在" + CONTENT_MIN + "-" + CONTENT_MAX + "之间";
    public static final int TITLE_MIN = 4;
    public static final int TITLE_MAX = 30;
    public static final String TITLE_VALIDATION_MSG= "标题长度在"+TITLE_MIN+"-"+TITLE_MAX+"之间";
    public static int IMAGES_AMOUNT_MAX=5;
    public static int IMAGES_AMOUNT_MIN=1;
    public static final String IMAGE_VALIDATION_MSG= "图片数量在"+IMAGES_AMOUNT_MIN+"-"+IMAGES_AMOUNT_MAX+"之间";

    public static int TAG_MAX=3;
    public static int TAG_MIN=1;
    public static final String TAG_VALIDATION_MSG= "标签数量在"+TAG_MIN+"-"+TAG_MAX+"之间";

    @TableId(type= IdType.INPUT)
    private Long id;
    private Long userId;
    private String content;
    private String title;
    private Integer starCount;
    private Integer pageView;
    private Integer commentCount;
    private String coverImageUrl;
    private Integer categoryId;

    private Date activityTime;
    private Date createTime;
    private Date updateTime;

    @TableLogic(delval = "1")
    boolean deleted;

    @TableField(exist = false)
    CmtyUser user;
    @TableField(exist = false)
    Category category;
    @TableField(exist = false)
    List<Integer> tagIds;
//    @TableField(exist = false)
//    List<Image> images;
    @TableField(exist = false)
    List<Comment> comments;
}
